test_runner/regress/test_wal_acceptor.py

```diff
@@ -2822,3 +2823,87 @@ def error_logged():
             cur.execute("select count(*) from t")
             # 2000 rows from first insert + 1000 from last insert
             assert cur.fetchone() == (3000,)
+
+
+def test_global_disk_usage_limit(neon_env_builder: NeonEnvBuilder):
+    """
+    Similar to `test_timeline_disk_usage_limit`, but test that the global disk usage circuit breaker
+    also works as expected. The test scenario:
+    1. Create a timeline and endpoint.
+    2. Start safekeeper with a very small global disk usage limit (1KB).
+    3. Write data to the timeline so that disk usage exceeds the limit.
+    4. Verify that the writes hang and the expected error message appears in the compute log.
+    5. Restart the safekeeper to clear the disk usage limit.
```

Comment:
It would be nice to have a variant of this test that simply clears the failpoint but doesn't restart the SK.

My understanding of this PR is that walproposer will eventually reconnect, not hit the disk usage check anymore, and things will resume by themselves, without an operator restarting safekeeper.

Reply:
Updated in [fixup: update test description](https://github.com/neondatabase/neon/pull/12605/commits/09fb7a6e2d03a909bc745b0c65f3b3f666e6a24c)